### PR TITLE
More resource_name cleanup

### PR DESCRIPTION
**Pull request by lamont-granquist**
Bit more cleanup surrounding resource_name

### CODE REVIEW

File: layouts/shortcodes/dsl_custom_resource_method_resource_name.md [hunk 1]
```diff
@@ -2,16 +2,14 @@ Use the `resource_name` method at the top of a custom resource to
...
-  # properties
-end
-```
\ No newline at end of file
+The `resource_name` is only used as a fallback name for display purposes.
+Even for display purposes the name used in recipe code which matches a
+`provides` statement is favored over the `resource_name` setting, so the
+`resource_name` has little effect.  In Chef Infra Client 16 and later
+the first `provides` in a resource declaration also sets the fall-back
+`resource_name`, so it recommended to not set it at all.  The `resource_name`
```
Comment (phiggins) [inline layouts/shortcodes/dsl_custom_resource_method_resource_name.md]:
```suggestion
`resource_name`, so it is recommended to not set it at all.  The `resource_name`
```

**Review by phiggins — APPROVED** [review]

File: content/definitions.md [hunk 2]
```diff
@@ -43,6 +43,8 @@ Definitions:
 -   Don't support `why-run` mode
 -   Can't report to Chef Automate
 -   Cannot be tested with ChefSpec
+-   There are known issues with the definition params hash where certain names (e.g. 'timeout')
```
Comment (IanMadd) [inline content/definitions.md]:
```suggestion
-   There are known issues with the definition parameters hash where certain names (e.g. 'timeout')
```

File: layouts/shortcodes/dsl_custom_resource_method_resource_name.md [hunk 3]
```diff
@@ -2,16 +2,14 @@ Use the `resource_name` method at the top of a custom resource to
...
-`:custom_name` is `web_httpd` then it may be used like this:
-
-``` ruby
-web_httpd 'name' do
-  # properties
-end
-```
\ No newline at end of file
+The `resource_name` is only used as a fallback name for display purposes.
+Even for display purposes the name used in recipe code which matches a
```
Comment (IanMadd) [inline layouts/shortcodes/dsl_custom_resource_method_resource_name.md]:
```suggestion
Even for display purposes, the name used in recipe code which matches a
```

File: layouts/shortcodes/dsl_custom_resource_method_resource_name.md [hunk 4]
```diff
@@ -2,16 +2,14 @@ Use the `resource_name` method at the top of a custom resource to
...
+`provides` statement is favored over the `resource_name` setting, so the
+`resource_name` has little effect.  In Chef Infra Client 16 and later
```
Comment (IanMadd) [inline layouts/shortcodes/dsl_custom_resource_method_resource_name.md]:
```suggestion
`resource_name` has little effect.  In Chef Infra Client 16 and later,
```

File: layouts/shortcodes/dsl_custom_resource_method_resource_name.md [hunk 5]
```diff
@@ -2,16 +2,14 @@ Use the `resource_name` method at the top of a custom resource to
...
+the first `provides` in a resource declaration also sets the fall-back
```
Comment (IanMadd) [inline layouts/shortcodes/dsl_custom_resource_method_resource_name.md]:
```suggestion
the first `provides` in a resource declaration also sets the fallback
```

**Review by IanMadd — COMMENTED** [review]
fixes for clarity

**Review by IanMadd — APPROVED** [review]

**Pull request merged**